torch/csrc/jit/codegen/cuda/codegen.cpp

```diff
@@ -1303,34 +1303,17 @@ class CudaKernelGenerator : private OptOutConstDispatch {
           indent() << "// Allocate global tensor " << varName(tv) << "\n";
           break;
         case MemoryType::Shared:
-          if (kir::ExpressionEvaluator::isConst(size)) {
-            // Static shared memory
-            //  Always align to 16B for tensorview buffers
-            //   with any vectorized access.
-            //  TODO:
-            //   This path will be less commonly exercised once we
-            //    start dynamically allocate all the tensors and
-            //    might be removed in a follow up.
-            auto va = kernel_->summary().vectorized_accesses;
-            if (va.count(tv)) {
-              indent() << "__align__(16) ";
-            } else {
-              indent();
-            }
-            code_ << "__shared__ " << buffer_dtype << " " << varName(tv) << "["
-                  << genInline(size) << "];\n";
-          } else {
-            // Align Offset Position
-            indent() << "offset = alignBufferSize(offset,"
-                     << dataTypeSize(buffer_dtype) << ");\n";
-            // Shared Memory Pointer
-            indent() << buffer_dtype << "* " << varName(tv)
-                     << " = reinterpret_cast<" << buffer_dtype << "*>"
-                     << "(array + offset);\n";
-            // Increment Offset Position
-            indent() << "offset += (" << genInline(size) << " * sizeof("
-                     << buffer_dtype << "));\n";
-          }
+          // Align Offset Position
+          indent() << "offset = alignBufferSize(offset,"
+                   // Always align to 128b / 16B
+                   << 16 << ");\n";
```

Comment:
This seems right to me, but this doesn't seem to agree: https://github.com/csarofeen/pytorch/blob/devel/torch/csrc/jit/codegen/cuda/executor.cpp#L330-L334

Can you make sure the dynamic shared memory calculation is correct taking into consideration alignment. We may just want to always add 16B per computed allocation in the above executor.cpp code for simplicity.

Reply:
It would be good to explicitly test the alignment, but I can't think of a good way to get a deterministic ordering of shared memory allocations so that the first will throw the alignment off for the second and we try to do vectorized loads with the second.

Reply:
I think it is correct since the no alignment padding case in executor.cpp is only used for static smems. It looks like we want to remove the static allocation path so I will just remove that part as well in the same follow up.

Reply:
It'd be great idea to add an alignment test. I could try to do a fusion with just a few tensors with different shapes and see if it covers the code paths.

Reply:
I see. Sorry I missed the alignment size on the last pass. Just updated so that executor always align to 128b as well.